core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -103,4 +103,7 @@ private TableProperties() {}
 
   public static final String WRITE_TARGET_FILE_SIZE_BYTES = "write.target-file-size-bytes";
   public static final long WRITE_TARGET_FILE_SIZE_BYTES_DEFAULT = Long.MAX_VALUE;
+
+  public static final String DISABLE_NULLABILITY_CHECK = "write.disable-nullability-check";
```

Comment:
I don't think this should be a table property. It is unlikely that you would want to ignore nullability for all writes to a table. The main problem is cases where a certain path produces a nullable field when the user knows it doesn't contain null, which is a more limited case. I think it makes sense for this to be a write option, and possibly an option that we get from the Spark environment. But I see no reason to make this apply across writes from different engines.